File: progSys/exec/python/exo4.py
import os
import subprocess
import sys

while True:
    pid = os.fork()

    if pid == 0:
        # Fils
        input_dir = input('Repertoire (Entrez "stop" pour quitter): ')

        if input_dir == "stop":
            os._exit(3)  # équivalent exit(3) en C
        else:
            # copie du répertoire courant vers le répertoire donné
            # équivalent à `cp -r . <input_dir>`
            subprocess.run(["cp", "-r", ".", input_dir])
            os._exit(0)  # fin du fils
    else:
        # Père
        pid_waited, status = os.wait()
        if os.WIFEXITED(status) and os.WEXITSTATUS(status) == 3:
            sys.exit(0)

File: progSys/exec/C/exo4.c
#include <unistd.h>
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/wait.h>

int main()
{
    char* input=malloc(10);
    pid_t pid;
    int a;
    while (1)
    {
        pid=fork();

        if (pid==0)
        {
            printf("Repertoire(Entrez \"stop\" pour quitter): ");
            scanf("%s",input);

            if (strcmp(input,"stop")==0)        
                exit(3);      
            else
            {
                printf("\n");
                execl("/usr/bin/cp","cp","-r",".",input,NULL);
            }
        }
        if (pid!=0)
        {
            wait(&a);
            if (WEXITSTATUS(a)==3)
                exit(0);
        }
    }
}
